packages/salesforce-adapter/src/transformers/transformer.ts

```diff
@@ -1658,9 +1663,10 @@ type CreateMetadataObjectTypeParams = Omit<ObjectTypeCtorParam, 'elemID'> & {
 }
 export const createMetadataObjectType = (
   params: CreateMetadataObjectTypeParams,
+  name?: string,
 ): MetadataObjectType =>
   new ObjectType({
-    elemID: new ElemID(SALESFORCE, params.annotations.metadataType),
+    elemID: new ElemID(SALESFORCE, name ?? params.annotations.metadataType),
```

Comment:
I see this change is only relevant for UTs and has no use-case in our code. Not a fan of changing code for the purpose of UTs. Also, doesn't this implementation makes more sense?

```suggestion
    elemID: new ElemID(SALESFORCE, METADATA_TYPES_TO_RENAME.get(params.annotations.metadataType) ?? params.annotations.metadataType),
```

Reply:
I just explicitly wrote it out in tests and reverted this change.